### PR TITLE
test(related-products): add dedupe meta reducer integration test

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[x] Other... Please describe: test
```

## What is the new behavior?
Added an integration test for ensuring that the deduping with the meta reducer behaves as expected.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/product/package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
     "@angular/common": "0.0.0-PLACEHOLDER",
     "@angular/core": "0.0.0-PLACEHOLDER",
     "@daffodil/core": "0.0.0-PLACEHOLDER",
+    "@daffodil/product": "0.0.0-PLACEHOLDER",
```
Comment (lderrickable) [inline libs/product/package.json:30]: was this a mistake?

Comment (griest024) [reply libs/product/package.json:30]: no that needs to be there. I can move it to a different PR though

Comment (lderrickable) [reply libs/product/package.json:30]: the product package is dependent on itself?

Comment (griest024) [reply libs/product/package.json:30]: oh.... haha yeah it was a mistake then

**Comment by lderrickable** [comment]
build fails

**Comment by griest024** [comment]
> build fails

I know, I don't know how to fix it. See the discussion in discord

**Pull request merged**